README.md
<!-- 
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages). 

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages). 
-->
# tbib_font_responsive

You can use the package to control your font easily.

you can control your font with google font package

## Features

You can use fonts dark or light like h1 - h1Dark


## Getting started

You need use flutter_screenutil library to use it
```yaml
dependencies:
  flutter:
    sdk: flutter
  # add flutter_screenutil
  flutter_screenutil: ^{latest version}
  tbib_font_responsive: ^{latest version}
```

## Usage


* lisenGoogleFont without choose h number
```dart

 TBIBFontStyle.lisenGoogleFont(GoogleFonts.roboto()); 
```

* lisenGoogleFont withchoose h number
```dart

   TBIBFontStyle.lisenGoogleFont(

      GoogleFonts.aBeeZee(textStyle: TBIBFontStyle.h2), 2);

      TBIBFontStyle.addCustomFont("BerthaMelanie", 3);
      
      you see ,2 ?? 2 meaning h2 in this library you have h1 to h6 only and b1 take style from h5 and b2 take style from h6
```
   * when use Pacakge google font use it in first code then edit your style

   ## handel screenUtil and tbib font responsive

  ```dart
      return ScreenUtilInit(
      designSize: const Size(360, 690),
      minTextAdapt: true,
      splitScreenMode: true,
      builder: (child) {
        return MaterialApp(
          title: 'Flutter Demo',
          theme: ThemeData(
            primarySwatch: Colors.blue,
          ),
          builder: (context, child) {
            ScreenUtil.init(context); // init screen Util
            log("old ${TBIBFontStyle.h1.fontSize}"); // see old font size
            ResponiveFont(); // start this class to start responsive font
             log("new ${TBIBFontStyle.h1.fontSize}"); // see new font size
              child: child!,
            );
          },
          home: const MyHomePage(title: ""), // start app
        );
      },
    );
```

CHANGELOG.md
## 2.0.1

* Now font responsive and how handel flutter_screenutil library.


lib/responsive_font.dart
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:tbib_font_responsive/style/font_style.dart';

class ResponiveFont {
  static bool _responsiveEnabled = false;

  ResponiveFont() {
    if (!_responsiveEnabled) {
      TBIBFontStyle.h1 =
          TBIBFontStyle.h1.copyWith(fontSize: TBIBFontStyle.h1.fontSize!.sp);
      TBIBFontStyle.h1Dark = TBIBFontStyle.h1Dark
          .copyWith(fontSize: TBIBFontStyle.h1Dark.fontSize!.sp);

      TBIBFontStyle.h2 =
          TBIBFontStyle.h2.copyWith(fontSize: TBIBFontStyle.h2.fontSize!.sp);
      TBIBFontStyle.h2Dark = TBIBFontStyle.h2Dark
          .copyWith(fontSize: TBIBFontStyle.h2Dark.fontSize!.sp);

      TBIBFontStyle.h3 =
          TBIBFontStyle.h3.copyWith(fontSize: TBIBFontStyle.h3.fontSize!.sp);
      TBIBFontStyle.h3Dark = TBIBFontStyle.h3Dark
          .copyWith(fontSize: TBIBFontStyle.h3Dark.fontSize!.sp);

      TBIBFontStyle.h4 =
          TBIBFontStyle.h4.copyWith(fontSize: TBIBFontStyle.h4.fontSize!.sp);
      TBIBFontStyle.h4Dark = TBIBFontStyle.h4Dark
          .copyWith(fontSize: TBIBFontStyle.h4Dark.fontSize!.sp);

      TBIBFontStyle.h5 =
          TBIBFontStyle.h5.copyWith(fontSize: TBIBFontStyle.h5.fontSize!.sp);
      TBIBFontStyle.h5Dark = TBIBFontStyle.h5Dark
          .copyWith(fontSize: TBIBFontStyle.h5Dark.fontSize!.sp);

      TBIBFontStyle.h6 =
          TBIBFontStyle.h6.copyWith(fontSize: TBIBFontStyle.h6.fontSize!.sp);
      TBIBFontStyle.h6Dark = TBIBFontStyle.h6Dark
          .copyWith(fontSize: TBIBFontStyle.h6Dark.fontSize!.sp);

      TBIBFontStyle.b1 =
          TBIBFontStyle.b1.copyWith(fontSize: TBIBFontStyle.b1.fontSize!.sp);
      TBIBFontStyle.b1Dark = TBIBFontStyle.b1Dark
          .copyWith(fontSize: TBIBFontStyle.b1Dark.fontSize!.sp);

      TBIBFontStyle.b2 =
          TBIBFontStyle.b2.copyWith(fontSize: TBIBFontStyle.b2.fontSize!.sp);
      TBIBFontStyle.b2Dark = TBIBFontStyle.b2Dark
          .copyWith(fontSize: TBIBFontStyle.b2Dark.fontSize!.sp);
      _responsiveEnabled = true;
    }
  }
}


lib/style/font_style.dart
import 'package:flutter/material.dart';

class TBIBFontStyle {
  static TextStyle h1 = const TextStyle(
      fontSize: 40,
      fontWeight: FontWeight.bold,
      height: 1.2,
      color: Colors.black);
  static TextStyle h2 = const TextStyle(
      fontSize: 32,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.black);
  static TextStyle h3 = const TextStyle(
      fontSize: 28,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.black);
  static TextStyle h4 = const TextStyle(
      fontSize: 24,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.black);
  static TextStyle h5 =
      const TextStyle(fontSize: 20, height: 1.2, color: Colors.black);
  static TextStyle h6 =
      const TextStyle(fontSize: 16, height: 1.2, color: Colors.black);
  static TextStyle b1 = const TextStyle(
      fontSize: 20,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.black);
  static TextStyle b2 = const TextStyle(
      fontSize: 16,
      fontWeight: FontWeight.w400,
      height: 1.2,
      color: Colors.black);

  static TextStyle h1Dark = const TextStyle(
      fontSize: 40,
      fontWeight: FontWeight.bold,
      height: 1.2,
      color: Colors.white);
  static TextStyle h2Dark = const TextStyle(
      fontSize: 32,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.white);
  static TextStyle h3Dark = const TextStyle(
      fontSize: 28,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.white);
  static TextStyle h4Dark = const TextStyle(
      fontSize: 24,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.white);
  static TextStyle h5Dark =
      const TextStyle(fontSize: 20, height: 1.2, color: Colors.white);
  static TextStyle h6Dark =
      const TextStyle(fontSize: 16, height: 1.2, color: Colors.white);
  static TextStyle b1Dark = const TextStyle(
      fontSize: 20,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.white);
  static TextStyle b2Dark = const TextStyle(
      fontSize: 16,
      fontWeight: FontWeight.w400,
      height: 1.2,
      color: Colors.white);

  static defaultFlutterStyle() {
    h1 = const TextStyle(
        fontSize: 96, fontWeight: FontWeight.w300, height: -1.5);
    h1Dark = h1.copyWith(color: Colors.white);
    h2 =
        const TextStyle(fontSize: 60, fontWeight: FontWeight.w300, height: -.5);
    h2Dark = h2.copyWith(color: Colors.white);
    h3 = const TextStyle(fontSize: 48, fontWeight: FontWeight.w400);
    h3Dark = h3.copyWith(color: Colors.white);
    h4 =
        const TextStyle(fontSize: 34, fontWeight: FontWeight.w300, height: .25);
    h4Dark = h4.copyWith(color: Colors.white);
    h5 = const TextStyle(fontSize: 24, fontWeight: FontWeight.w300);
    h5Dark = h5.copyWith(color: Colors.white);
    h6 =
        const TextStyle(fontSize: 20, fontWeight: FontWeight.w500, height: .15);
    h6Dark = h6.copyWith(color: Colors.white);
    b1 = const TextStyle(fontSize: 16, fontWeight: FontWeight.w400, height: .5);
    b1Dark = b1.copyWith(color: Colors.white);
    b2 =
        const TextStyle(fontSize: 14, fontWeight: FontWeight.w400, height: .15);
    b2Dark = b2.copyWith(color: Colors.white);
  }

  static lisenGoogleFont(TextStyle googleFontsName, {int? h}) {
    if (h != null) {
      switch (h) {
        case 1:
          {
            h1Dark = h1 = h1.copyWith(fontFamily: googleFontsName.fontFamily);
            h1Dark = h1Dark.copyWith(color: Colors.white);
            return;
          }
        case 2:
          {
            h2Dark = h2 = h2.copyWith(fontFamily: googleFontsName.fontFamily);
            h2Dark = h2Dark.copyWith(color: Colors.white);
            return;
          }
        case 3:
          {
            h3Dark = h3 = h3.copyWith(fontFamily: googleFontsName.fontFamily);
            h3Dark = h3Dark.copyWith(color: Colors.white);

            return;
          }
        case 4:
          {
            h4Dark = h4 = h4.copyWith(fontFamily: googleFontsName.fontFamily);
            h4Dark = h4Dark.copyWith(color: Colors.white);

            return;
          }
        case 5:
          {
            h5Dark = h5 = b1 =
                b1Dark = h5.copyWith(fontFamily: googleFontsName.fontFamily);
            h5Dark = h5Dark.copyWith(color: Colors.white);
            b1Dark = b1Dark.copyWith(color: Colors.white);

            return;
          }
        case 6:
          {
            h6 = h6Dark = b2 =
                b2Dark = h6.copyWith(fontFamily: googleFontsName.fontFamily);
            h6Dark = h6Dark.copyWith(color: Colors.white);
            b2Dark = b2Dark.copyWith(color: Colors.white);

            return;
          }
        default:
          {
            return null;
          }
      }
    } else {
      h1Dark = h1 = googleFontsName.copyWith(
          fontSize: 40, fontWeight: FontWeight.bold, height: 1.2);
      h1Dark = h1Dark.copyWith(color: Colors.white);

      h2Dark = h2 = googleFontsName.copyWith(
          fontSize: 32, fontWeight: FontWeight.w600, height: 1.2);
      h2Dark = h2Dark.copyWith(color: Colors.white);

      h3Dark = h3 = googleFontsName.copyWith(
          fontSize: 28, fontWeight: FontWeight.w600, height: 1.2);
      h3Dark = h3Dark.copyWith(color: Colors.white);

      h4Dark = h4 = googleFontsName.copyWith(
          fontSize: 24, fontWeight: FontWeight.w500, height: 1.2);
      h4Dark = h4Dark.copyWith(color: Colors.white);

      h5Dark = h5 =
          b1 = b1Dark = googleFontsName.copyWith(fontSize: 20, height: 1.2);
      h5Dark = h5Dark.copyWith(color: Colors.white);
      b1Dark = b1Dark.copyWith(color: Colors.white);

      h6 = h6Dark =
          b2 = b2Dark = googleFontsName.copyWith(fontSize: 16, height: 1.2);
      h6Dark = h6Dark.copyWith(color: Colors.white);
      b2Dark = b2Dark.copyWith(color: Colors.white);
    }
  }

  static void addCustomFont(String yourFontFamily, {int? h}) {
    if (h != null) {
      switch (h) {
        case 1:
          {
            h1 = h1.copyWith(fontFamily: yourFontFamily);
            h1Dark = h1Dark.copyWith(fontFamily: yourFontFamily);

            break;
          }
        case 2:
          {
            h2 = h2.copyWith(fontFamily: yourFontFamily);
            h2Dark = h2Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 3:
          {
            h3 = h3.copyWith(fontFamily: yourFontFamily);
            h3Dark = h3Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 4:
          {
            h4 = h4.copyWith(fontFamily: yourFontFamily);
            h4Dark = h4Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 5:
          {
            h5 = h5.copyWith(fontFamily: yourFontFamily);
            h5Dark = h5Dark.copyWith(fontFamily: yourFontFamily);
            b1 = b1.copyWith(fontFamily: yourFontFamily);
            b1Dark = b1Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 6:
          {
            h6 = h6.copyWith(fontFamily: yourFontFamily);
            h6Dark = h6Dark.copyWith(fontFamily: yourFontFamily);
            b2 = b2.copyWith(fontFamily: yourFontFamily);
            b2Dark = b2Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        default:
          {
            break;
          }
      }
    } else {
      h1 = h1.copyWith(fontFamily: yourFontFamily);
      h1Dark = h1Dark.copyWith(fontFamily: yourFontFamily);
      h2 = h2.copyWith(fontFamily: yourFontFamily);
      h2Dark = h2Dark.copyWith(fontFamily: yourFontFamily);
      h3 = h3.copyWith(fontFamily: yourFontFamily);
      h3Dark = h3Dark.copyWith(fontFamily: yourFontFamily);
      h4 = h4.copyWith(fontFamily: yourFontFamily);
      h4Dark = h4Dark.copyWith(fontFamily: yourFontFamily);
      h5 = h5.copyWith(fontFamily: yourFontFamily);
      h5Dark = h5Dark.copyWith(fontFamily: yourFontFamily);
      h6 = h6.copyWith(fontFamily: yourFontFamily);
      h6Dark = h6Dark.copyWith(fontFamily: yourFontFamily);
      b1 = b1.copyWith(fontFamily: yourFontFamily);
      b1Dark = b1Dark.copyWith(fontFamily: yourFontFamily);
      b2 = b2.copyWith(fontFamily: yourFontFamily);
      b2Dark = b2Dark.copyWith(fontFamily: yourFontFamily);
    }
  }
}


example/lib/main.dart
import 'dart:developer';

import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:google_fonts/google_fonts.dart';

import 'package:tbib_font_responsive/tbib_font_responsive.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  // use font google in h1 and h1Dark
  TBIBFontStyle.lisenGoogleFont(GoogleFonts.aBeeZee(), h: 1);

  // use custom font in h3 and h3Dark only
  TBIBFontStyle.addCustomFont("BerthaMelanie", h: 3);

  // use custom font in all
  TBIBFontStyle.addCustomFont("BerthaMelanie");

  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return ScreenUtilInit(
      designSize: const Size(360, 690),
      minTextAdapt: true,
      splitScreenMode: true,
      builder: (child) {
        return MaterialApp(
          title: 'Flutter Demo',
          theme: ThemeData(
            primarySwatch: Colors.blue,
          ),
          builder: (context, child) {
            ScreenUtil.init(context);
            log("old ${TBIBFontStyle.h1.fontSize}");
            ResponiveFont();
            return child!;
          },
          home: const MyHomePage(title: ""),
        );
      },
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool isDark = false;

  @override
  void initState() {
    // TODO: implement initState
    super.initState();
    log("new ${TBIBFontStyle.h1.fontSize}");
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
        actions: [
          IconButton(
              onPressed: () {
                setState(() {
                  isDark = !isDark;
                });
              },
              icon: const Icon(Icons.brightness_4))
        ],
      ),
      body: Container(
          color: isDark ? Colors.black : null,
          child: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Text(
                  "Text custom font",
                  style: isDark ? TBIBFontStyle.h1Dark : TBIBFontStyle.h1,
                ),
                const SizedBox(
                  height: 20,
                ),
                Text(
                  "Text color mode",
                  style: isDark ? TBIBFontStyle.h2Dark : TBIBFontStyle.h2,
                ),
                const SizedBox(
                  height: 20,
                ),
                Text(
                  "Text color mode",
                  style: isDark ? TBIBFontStyle.h3Dark : TBIBFontStyle.h3,
                ),
              ],
            ),
          )),
    );
  }
}
